src/settings/LocationLocations/remoteStorageDetails.test.js

```diff
@@ -29,13 +24,31 @@ const renderRemoteStorageDetails = ({
 ));
 
 describe('RemoteStorageDetails', () => {
+  beforeEach(() => {
+    useRemoteStorageApi.mockImplementation(() => ({ remoteMap: mockRemoteMap, configurations: mockConfigurations }));
+  });
+
   it('should render remote storage name and workflow preference if location have them', () => {
     renderRemoteStorageDetails({ locationId: 'locationWithDetails' });
 
     expect(screen.getByText(/remoteStorage/)).toBeVisible();
     expect(screen.getByText(/returning-workflow.title/)).toBeVisible();
   });
 
+  it('should render name for location with details', () => {
+    renderRemoteStorageDetails({ locationId: 'locationWithDetails' });
+
+    expect(screen.getByText('RS2')).toBeVisible();
+  });
+
+  it('should not render name if state is broken', () => {
+    useRemoteStorageApi.mockImplementation(() => ({ remoteMap: undefined, configurations: undefined }));
+
+    renderRemoteStorageDetails({ locationId: 'locationWithDetails' });
+
+    expect(screen.queryByText('RS2')).not.toBeInTheDocument();
+  });
+
```

Comment:
Does it cover the bug in scope?
I mean, if we rollback that faulty `useMemo` thing, will the tests break?